### DISCUSSION TITLE
Regarding access to arbitrary front matter items in child pages from a category index page

### DISCUSSION THREAD

**Discussion by allenscha** [facebook/docusaurus#8871] [Q&A] [answered]
I found out that customProps (`sidebar_custom_props`) has been added in order to allow access to front matter items in child pages from a category index page

https://github.com/facebook/docusaurus/issues/6590

and the front matter items can be accessed inside the Markdown page itself as follows:

```
{frontMatter.version}
```

But is there a way to access front matter items other than `sidebar_custom_props`, such as `tags`, in child pages from a category index page?

I'm trying to implement a filter function in a category index page based on label(s) of the child pages. However, since `sidebar_custom_props` only allows pairs of a key and a value, I find it difficult to use the field for multi-label filtering. So I'm trying to find if I can use `tags` or some other field that allows list of labels instead.

**Comment by slorber** [comment, marked as answer]
Not sure to understand, please create a repro with an example of what you are trying to do.

> However, since sidebar_custom_props only allows pairs of a key and a value, I find it difficult to use the field for multi-label filtering

`sidebar_custom_props` allow key value pairs but the values can be more complex types, you can pass a Yaml array as value for example.

Tags are not added to the sidebar item props, so if you want to use tags for your filtering, you'd have to "duplicate" them:

```md
---
id: doc-with-tags
title: A doc with tags
tags:
  - Demo
  - Getting started
sidebar_custom_props:  
  tags:
    - Demo
    - Getting started
---
```

In the future a `createFrontMatter` API would allow you to do that automatically without duplication (see https://github.com/facebook/docusaurus/issues/5568)

**Reply by allenscha** [reply]
Thanks for your answer. I was not aware that more complex types can be passed to the values of `sidebar_custom_props`, judging only from the examples on the test case code. Probably it would be better to elaborate the documentation regarding this point:
https://docusaurus.io/docs/api/plugins/@docusaurus/plugin-content-docs#sidebar_custom_props

**Reply by slorber** [reply]
Thanks @allenscha , I improved our docs here to make this clearer: https://github.com/facebook/docusaurus/pull/8888
